### PR TITLE
[KHWEBA-5].1 Bar chart

### PR DESCRIPTION
**Pull request by gift-a**
image

### CODE REVIEW

**Comment by ihor7016** [comment]
image

* please use colors similar to ones that are on the image (transparent effect is also nice to have)
* probably it is ok to use grid lines in chart
* make chart smaller
* bars should be thinner
* remove the title from the chart (Trend)
* make card's padding smaller

File: src/components/bar-chart/bar-chart.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  querySelectors() {
+    this.barChartCtx = this.mountPoint.querySelector(".bar-chart__visual");
+  }
+
+  createBarChart() {
+    Chart.defaults.global.defaultFontFamily = "'Roboto', sans-serif";
```
Comment (ihor7016) [inline src/components/bar-chart/bar-chart.js]: is it possible to set these `global defaults` only for this chart ?

**Review by Vladyslav-Plakhuta — DISMISSED** [review]

**Comment by gift-a** [comment]
what do you mean "make chart smaller"?
should it have card width less than ~50%?

**Comment by gift-a** [comment]
image
image

**Comment by gift-a** [comment]
done

**Comment by ihor7016** [comment]
* please bring back solid colors (no transparency, no borders) but take them from material (and not that ones that you used before -- they look ugly)
* add grid lines to the chart (not material components grid -- chart grid lines)
* make padding from the left and right a bit more bigger
* ensure that your layout will fit 2 charts even on smaller screen

**Comment by gift-a** [comment]
> please bring back solid colors (no transparency) but take them from material (and not that ones that you used before -- they look ugly)

The colors I've used before were from material:
image

this font color is the main for our web page

**Comment by ihor7016** [comment]
I believe you, but these red and green colors look ugly please choose other

image

**Comment by gift-a** [comment]
oh, sorry. it seems i didn't push

**Comment by gift-a** [comment]
so I should not change font colors

**Comment by gift-a** [comment]
if you don't like these colors, you can choose it here
https://material.io/color/#!/?view.left=0&view.right=0&primary.color=4CAF50

**Comment by ihor7016** [comment]
colors nice
but according to mockup your chart should be in the right card

image

**Comment by ihor7016** [comment]
+ on the small screen second card hides for some reason
it should stay on the same line

image

**Comment by gift-a** [comment]
I didn't manage to make it with flexbox. in unknown for me reason these cards become very small

**Comment by ihor7016** [comment]
it does not matter what you use -- please fix the behavior on the small screens -- both cards should stay on the same line

image

**Comment by ihor7016** [comment]
if they are really too small -- make each card stay on new line on small screens

**Comment by ihor7016** [comment]
Ok let it be not responsive.I've just merged Kostya's pull request -- just resolve conflicts and replace second chart with yours chart

**Comment by gift-a** [comment]
ready to merge

File: src/components/bar-chart/bar-chart.scss [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.bar-chart {
```
Comment (ihor7016) [inline src/components/bar-chart/bar-chart.scss]: please extract `bar chart` and `pie chart` into separate style-only  (BEM) component and reuse these styles in both charts -- and the remove these specific bar-chart and pie-chart files

**Comment by gift-a** [comment]
tried to resolve problem with flex. could you pls have a look on it?
https://github.com/ihor7016/buxy-vanilla/blob/KHWEBA-5.1_positioning/src/components/bar-chart/bar-chart.scss

**Comment by ihor7016** [comment]
why do you need flex inside chart?

**Comment by gift-a** [comment]
because flex provides vertical positioning when you don't know heights of inner and outer elems

File: src/components/bar-chart/bar-chart.scss (hunk 2 above)
Comment (gift-a) [reply src/components/bar-chart/bar-chart.scss]: separate folder for one chart.scss style?

**Comment by ihor7016** [comment]
image

http://www.chartjs.org/docs/latest/general/responsive.html#important-note

**Comment by ihor7016** [comment]
please just make it work 

and

extract bar chart and pie chart into separate style-only (BEM) component and reuse these styles in both charts -- and the remove these specific bar-chart and pie-chart files

**Comment by gift-a** [comment]
not ready yet

**Comment by gift-a** [comment]
ready to merge

**Review by ihor7016 — APPROVED** [review]

**Pull request merged**